src/legacy.rs

```diff
@@ -0,0 +1,194 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{
+    aesgcm::{AesGcmEceWebPush, AesGcmEncryptedBlock},
+    common::WebPushParams,
+    crypto::EcKeyComponents,
+    error::*,
+};
+
+/// Encrypt a block using legacy AESGCM encoding.
+///
+/// param remote_pub &[u8] - The remote public key
+/// param remote_auth &u8 - The remote authorization token
+/// param salt &[u8] - The locally generated random salt
+/// param data &[u8] - The data to encrypt
+///
+pub fn encrypt_aesgcm(
+    remote_pub: &[u8],
+    remote_auth: &[u8],
+    salt: &[u8],
```

Comment:
I think ideally we would not accept a `salt` argument here, since it just creates an opportunity for the caller to do something wrong when generating it. However, the existing top-level `encrypt` function currently takes a `salt` argument, so it makes sense to mirror that here 👍 

I'm mostly saying this out loud as a reminder to myself that a future API change should remove `salt` from both of these functions...

Reply:
Well it does provide a nice way of testing the function result against the IETF versions.
If removed, those tests would need to be done on a lower-level layer, possibly recoding part of the function.

A suggestion I could make would be to have private versions of these functions, with the `salt` argument, that could be tested, and a public version that just generates a random salt and calls the underlying testable and private function ... ?

Reply:
Indeed, it's very convenient for the tests. Figuring out how to do better here can be a problem for our future selves.